scenes/tower/tower.gd

```diff
@@ -35,35 +39,61 @@ func _ready() -> void:
 	_set_tower_attributes(tower_resource)
 	_set_attack_radius(tower_resource.attack_radius)
 	sprite.texture = tower_resource.texture
-	
-	
+	fire_timer.wait_time = tower_resource.fire_rate
+
+
 func _draw() -> void:
 	var blue_color: Color = Color.blue
 	blue_color.a = alpha_area
 	draw_circle(Vector2.ZERO, attack_radius, blue_color)
-	
-	
+
+
+func _on_AttackRangeShowArea_mouse_entered() -> void:
+	alpha_area = alpha_area_show
+	update()
+
+
+func _on_AttackRangeShowArea_mouse_exited() -> void:
+	alpha_area = alpha_area_hide
+	update()
+
+func _on_Area2D_body_entered(body):
+	if body is Enemy:
+		change_state(2)
```

Comment:
Can you replace the integers with the enum key? Like here (and in next occurrences) it's possible to use `TOWER_STATES.ATTACKING`

Reply:
Solved in d4f6d47!